spec/datadog/core/configuration/config_helper_spec.rb

```diff
@@ -0,0 +1,407 @@
+      let(:name) { 'DD_TRACE_ENABLED' }
+      let(:env_vars) { { 'DD_TRACE_ENABLED' => 'true' } }
+      let(:supported_configurations) { { 'DD_TRACE_ENABLED' => { version: ['A'] } } }
+
+      it 'returns the environment variable value' do
+        is_expected.to eq('true')
+      end
+    end
+
+    context 'when the environment variable is not set' do
+      let(:name) { 'DD_TRACE_ENABLED' }
+      let(:env_vars) { {} }
+      let(:supported_configurations) { { 'DD_TRACE_ENABLED' => { version: ['A'] } } }
+
+      it 'returns nil' do
+        is_expected.to be_nil
+      end
+
+      context 'when a default value is provided' do
+        subject { described_class.get_environment_variable(name, 'default', env_vars: env_vars) }
+
+        it 'returns the default value' do
+          is_expected.to eq('default')
+        end
+      end
+    end
+
+    context 'when the environment variable has an alias' do
+      let(:name) { 'DD_SERVICE' }
+      let(:env_vars) { { 'OTEL_SERVICE_NAME' => 'my-service' } }
+      let(:supported_configurations) { { 'DD_SERVICE' => { version: ['A'] } } }
+      let(:aliases) { { 'DD_SERVICE' => ['OTEL_SERVICE_NAME'] } }
+
+      it 'returns the alias value when the main env var is not set' do
+        is_expected.to eq('my-service')
+      end
+
+      context 'when both main and alias are set' do
+        let(:env_vars) { { 'DD_SERVICE' => 'main-service', 'OTEL_SERVICE_NAME' => 'alias-service' } }
+
+        it 'returns the main environment variable value' do
+          is_expected.to eq('main-service')
+        end
+      end
+    end
+
+    context 'when Datadog::CI is defined' do
+      before do
+        stub_const('Datadog::CI', Module.new)
+      end
+
+      let(:name) { 'DD_UNSUPPORTED_VAR' }
+      let(:env_vars) { { 'DD_UNSUPPORTED_VAR' => 'value' } }
+      let(:supported_configurations) { {} }
+
+      # For now, datadog-ci-rb is not supported and we don't want to break it.
+      it 'returns the environment variable value even if unsupported' do
+        is_expected.to eq('value')
+      end
+    end
+
+    context 'when environment variable starts with DD_ but is not supported' do
+      let(:name) { 'DD_UNSUPPORTED_VAR' }
+      let(:env_vars) { { 'DD_UNSUPPORTED_VAR' => 'value' } }
+      let(:supported_configurations) { {} } # Override to make it unsupported
+
+      context 'when not in test environment' do
+        around do |example|
+          described_class.instance_variable_set(:@raise_on_unknown_env_var, nil)
+
+          example.run
+
+          described_class.instance_variable_set(:@raise_on_unknown_env_var, true)
+        end
+
+        it 'returns nil' do
+          is_expected.to be_nil
+        end
+      end
+
+      context 'when in test environment' do
+        it 'raises an error for unsupported DD_ variables' do
+          expect { subject }.to raise_error(RuntimeError, /Missing DD_UNSUPPORTED_VAR env\/configuration/)
+        end
+      end
+    end
```

Comment:
Consider having `@raise_on_unknown_env_var` be an instance variable of `ConfigHelper`. This way, it could be one more thing that we can set in the constructor, and thus no need to touch the global state here!